lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -395,17 +395,16 @@ def read_streams
           # read a Little Endian 32-bit integer for length of response
           expected_response_length = pipe.sysread(4).unpack('V').first
 
-          if expected_response_length == 0
-            nil
-          else
-            # reads the expected bytes as a binary string or fails
-            buffer = ""
-            # Reads in the pipe data 8K, or less, at a time
-            while (buffer.length < expected_response_length)
-              buffer << pipe.sysread([expected_response_length - buffer.length, 8192].min)
-            end
-            buffer
+          next nil if expected_response_length == 0
+          # reads the expected bytes as a binary string or fails
+          buffer = ""
+          # sysread may not return all of the requested bytes due to buffering or the
+          # underlying IO system. Keep reading from the pipe until all the bytes are read
+          loop do
+            buffer.concat(pipe.sysread(expected_response_length - buffer.length))
+            break if buffer.length == expected_response_length
```

Comment:
Infinite loops are always a little risky... are there other conditions that can / should exit? Should we change `==` to `>=`?

Reply:
agreed

Reply:
Done!